primitives/src/circuit/merkle_tree/mod.rs

```diff
@@ -36,9 +36,10 @@ mod rescue_merkle_tree;
 /// circuit.enforce_merkle_proof(leaf_var, path_vars, root_var).unwrap();
 /// assert!(circuit.check_circuit_satisfiability(&[]).is_ok());
 /// ```
-pub trait MerkleTreeGadget<M>
+pub trait MerkleTreeGadget<M, D>
 where
     M: MerkleTreeScheme,
+    D: DigestAlgorithmGadget<M> + ?Sized,
```

Comment:
Why does it have to be `?Sized`?

Reply:
Usually it's because we use it in dynamic trait objects somewhere